apps/aechannel/src/aesc_fsm.erl

```diff
@@ -233,9 +233,18 @@ respond(Port, #{} = Opts0) ->
         ok ->
             aesc_fsm_sup:start_child([#{ port => Port
                                        , opts => Opts }]);
-        {error, _Reason} = Err -> Err
+        {error, _Reason} = Err ->
+            Err
     ?CATCH_LOG(_E)
-        {error, _E}
+            {error, _E}
+    end.
+
+check_limits(Opts) ->
+    case maps:is_key(existing_channel_id, Opts) of
```

Comment:
What if this value is not at least a valid channel id? E.g. `undefined`?

Reply:
Then the reestablish will fail, except a bit later. It might be noted that `aesc_fsm:init_/1` also uses `maps:is_key(existing_channel_id, Opts1)` to determine whether a reestablish should be attempted.